postgresql_proxy/proxy.py

```diff
@@ -51,50 +61,78 @@ def __create_pg_connection(self, address, context):
         pg_sock.connect((redirect_config.host, redirect_config.port))
         pg_sock.setblocking(False)
 
-        events = selectors.EVENT_READ | selectors.EVENT_WRITE
```

Comment:
Not sure about this change. I think we would still want to listen to write events from the socket, in order to forward the bidirectional communication, right? I briefly tested this with our integration tests in -ext - will add more details and steps to reproduce there. 👍

Reply:
Update: On second look, I think I understand now why you've removed EVENT_WRITE from the selector - since we can assume that the connection is always triggered by a READ event (either on the proxy connection from the client, or on the PG connection coming from the server), and `service_connection(..)` then first performs a `recv` operation on the connection, and then a `send` operation to forward the data to the receiving end. 👍

Reply:
The EVENT_WRITE and EVENT_READ are there to indicate the socket is "ready for write" and "ready to be read" respectively, in the sense the write buffer is not full and the read buffer has something in it. So the "EVENT_WRITE" would be ready all the time and we would busy loop on it. But yes, we're basically reading on both side!